token-client/src/main/java/com/sap/cloud/security/xsuaa/client/OAuth2TokenKeyService.java

```diff
@@ -50,4 +50,26 @@ public interface OAuth2TokenKeyService {
 	default String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId) throws OAuth2ServiceException {
 		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId);
 	}
+
+	/**
+	 * Requests token web key set from IAS OAuth Server.
+	 *
+	 * @param tokenKeysEndpointUri
+	 *            the token endpoint URI (jku).
+	 * @param tenantId
+	 *            the unique identifier of the tenant. Obligatory parameter in context of
+	 *            multi-tenant IAS applications to make sure that the tenant id
+	 *            belongs to the IAS tenant.
+	 * @param clientId
+	 * 				clientId from the service binding
+	 * @param azp
+	 * 				azp from token
+	 * @return list of JSON Web Token (JWT) keys as
+	 *         JSON string.
+	 * @throws OAuth2ServiceException
+	 *             in case of an error during the http request.
+	 */
+	default String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId, @Nullable String azp) throws OAuth2ServiceException {
```

Comment:
refactor this method as well, same as above in the OAuth2TokenKeyServiceWithCache.java